u19_pipeline/puffs.py

```diff
@@ -0,0 +1,82 @@
+"""This module defines tables in the schema ahoag_puffs_lab_demo"""
+
+import datajoint as dj
+from . import lab, acquisition, task
+
+schema = dj.schema(dj.config['database.prefix'] + 'puffs')
+
+@schema
+class PuffsCohort(dj.Manual):
+    definition = """
+    -> lab.User         
+    project_name         : varchar(64)                  # Corresponds to the path on bucket /puffs/netid/project_name/cohortX/ ...
+    cohort               : varchar(64)                  # Corresponds to the path on bucket /puffs/netid/project_name/cohortX/ ...
+    ---
+    """
+
+@schema
+class PuffsFileAcquisition(dj.Manual):
+    definition = """
+    -> PuffsCohort
+    rig                  : tinyint                      # an integer that describes which rig was used. 0 corresponds to location = "pni-ltl016-05", 1 corresponds to location = "wang-behavior"
+    h5_filename          : varchar(256)                 # The full path name, e.g. "data.h5" or "data_compressed_XX.h5" for the h5 behavior data file
+    ---
+    ingested             : boolean                      # A flag for whether this file has already been ingested.  
+    """
+
+"""This module defines tables in the schema ahoag_puffs_behavior_demo"""
+
+
+@schema
+class PuffsSession(dj.Manual):
+    definition = """
+    -> acquisition.Session
+    ---
+    session_params = NULL       : blob                         # The parameters for this session, e.g. phase_durations, whether puffs are on, etc... 
+    rig                         : tinyint                      # an integer that describes which rig was used. 0 corresponds to location = "pni-ltl016-05", 1 corresponds to location = "wang-behavior"
+    notes = ''                  : varchar(1024)                # notes recorded by experimenter during the session
+    stdout = NULL               : blob                         # stdout for the GUI during the session
+    stderr = NULL               : blob                         # stderr for the GUI during the session
+    sync = NULL                 : blob                         # At the start of the session, the software runs multiple python processes. This column contains the times of these processes in seconds 
+    """
+
+    class Trial(dj.Part):
+    	definition = """
+        -> PuffsSession
+        trial_idx            : int                          # trial index, keep the original number in the file
+        ---
+		-> task.TaskLevelParameterSet.proj(level="level")   # The difficulty level of the trial
```

Comment:
Why do we need the `proj()` here?

Reply:
good point we don't. It was left over from when I was renaming level, but I changed it back.